### PR TITLE
logical_range_filter: separate window from logical_range_filter into an independent file

### PR DESCRIPTION
**Pull request by HashidaTKS**
Separate `window` from `logical_range_filter` into an independent file.

This is a preparation for modification of `logical_count` to un-refer shards immediately like `logical_range_filter`.

### CODE REVIEW

File: plugins/sharding/window.rb [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        [@year, @month, @day, @hour, @minute, @second, @microsecond]
+      end
+      
+      def <=>(other)
+        (values + [@border == :include ? 1 : 0]) <=>
+          (other.values + [other.border == :include ? 1 : 0])
+      end
+    end
+  end
+end
```
Comment (kou) [inline plugins/sharding/window.rb:236]: Could you add a new line?

**Comment by kou** [comment]
Could you also update `plugins/sharding/sources.am`?

File: plugins/sharding/window.rb (hunk 1 above)
Comment (HashidaTKS) [reply plugins/sharding/window.rb]: Sorry for forgetting it so many times...
I have fixed it.

**Comment by HashidaTKS** [comment]
@kou 

Thank you for the review.

> Could you also update plugins/sharding/sources.am?

I have added `window.rb` into `plugins/sharding/sources.am`.

**Comment by kou** [comment]
Thanks.

**Pull request merged**